src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -200,24 +200,24 @@ EPOCHS = 2; // make test faster in Debug
       };
       goldSP.setSparse(deterministicSP);
 
       SDR goldSPlocal({COLS});
       const SDR_sparse_t deterministicSPlocal{
-        12, 13, 71, 72, 75, 78, 82, 85, 131, 171, 182, 186, 189, 194, 201, 263, 277, 287, 308, 319, 323, 337, 339, 365, 407, 423, 429, 432, 434, 445, 493, 494, 502, 508, 523, 542, 554, 559, 585, 586, 610, 611, 612, 644, 645, 647, 691, 698, 699, 701, 702, 707, 777, 809, 810, 811, 833, 839, 841, 920, 923, 928, 929, 935, 955, 1003, 1005, 1073, 1076, 1094, 1095, 1114,1115, 1133, 1134, 1184, 1203, 1214, 1232, 1233, 1244, 1253, 1268, 1278, 1291, 1294, 1306, 1309, 1331, 1402, 1410, 1427, 1434, 1442, 1463, 1508, 1512, 1514, 1515, 1518, 1561, 1564, 1590, 1623, 1626, 1630, 1647, 1651, 1691, 1694, 1729, 1745, 1746, 1760, 1797, 1804, 1805, 1812, 1827, 1858, 1860, 1861, 1862, 1918, 1956, 1961, 1965, 1971, 1975, 1994, 2012
+        13, 62, 71, 72, 73, 78, 80, 134, 140, 167, 169, 179, 189, 194, 261, 263, 268, 269, 308, 323, 328, 337, 339, 365, 407, 423, 425, 432, 434, 443, 493, 494, 508, 512, 514, 520, 585, 598, 601, 610, 630, 644, 645, 673, 675, 691, 701, 707, 748, 749, 777, 809, 811, 833, 838, 841, 853, 889, 906, 921, 926, 928, 952, 958, 967, 989, 1005, 1076, 1089, 1095, 1114, 1115,1120, 1133, 1146, 1181, 1184, 1196, 1203, 1217, 1249, 1252, 1253, 1263, 1282, 1291, 1306, 1309, 1331, 1337, 1401, 1402, 1410, 1434, 1462, 1469, 1487, 1494, 1508, 1512, 1518, 1547, 1563, 1564, 1623, 1624, 1626, 1651, 1672, 1677, 1693, 1694, 1745, 1746, 1750, 1760, 1768, 1802, 1805, 1831, 1858, 1861, 1869, 1880, 1889, 1929, 1950, 1956, 1961, 1994, 2002, 2011, 2012, 2027
       };
       goldSPlocal.setSparse(deterministicSPlocal);
 
       SDR goldTM({COLS});
       const SDR_sparse_t deterministicTM{
-        62, 77, 85, 322, 340, 432, 952, 1120, 1488, 1502, 1512, 1518, 1547, 1627, 1633, 1668, 1727, 1729, 1797, 1803, 1805, 1812, 1858, 1859, 1896, 1918, 1923, 1925, 1929, 1931, 1939, 1941, 1942, 1944, 1950, 1953, 1955, 1956, 1965, 1966, 1967, 1968, 1974, 1980, 1987, 1996, 2006, 2008, 2011, 2027, 2030, 2042, 2046
+        17, 37, 62, 72, 73, 93, 134, 261, 268, 269, 301, 309, 323, 331, 337, 338, 432, 493, 502, 523, 532, 542, 645, 754, 833, 921, 952, 1102, 1114, 1507, 1508, 1512, 1627, 1693, 1858, 1861, 1870, 1917, 1925, 1927, 1929, 1937, 1938, 1941, 1943, 1944, 1945, 1946, 1947, 1950, 1952, 1955, 1956, 1959, 1960, 1964, 1966, 1969, 1971, 1974, 1975, 1978, 1979, 1987, 1990, 1991,1995, 1996, 1998, 1999, 2002, 2012, 2027, 2040, 2042
       };
       goldTM.setSparse(deterministicTM);
 
-      const float goldAn    = 0.627451f;
-      const float goldAnAvg = 0.407265f;
+      const float goldAn    = 0.5f;
+      const float goldAnAvg = 0.386638f;
```

Comment:
btw, this change slightly improves anomaly here and also translates to improvement on NAB dataset